### PR TITLE
Add basic authentication support

### PR DESCRIPTION
**Pull request by arturo-seijas**
Applicable spec: N/A

### Overview

* Add basic authentication for the present and cleanup endpoints
* Also remove the corresponding views
* Remove some leftover files

### Rationale

N/A

### Module Changes

N/A

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by mthaddon — APPROVED** [review]
Looks great, thanks!

File: httprequest_lego_provider/basicauth.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+        Returns:
+            an HTTP response containing the view if authentication succeeds, 401 otherwise.
+        """
+        if "HTTP_AUTHORIZATION" in request.META:
+            auth = request.META["HTTP_AUTHORIZATION"].split()
+            if len(auth) == 2 and auth[0].lower() == "basic":
+                uname, passwd = base64.b64decode(auth[1]).decode("ascii").split(":", 1)
+                user = authenticate(username=uname, password=passwd)
+                if user is not None and user.is_active:
```
Comment (amandahla) [inline httprequest_lego_provider/basicauth.py:47]: Out of curiosity: will return 403 if user is not authorized?

**Review by amandahla — APPROVED** [review]
LGTM

File: httprequest_lego_provider/basicauth.py (hunk 1 above)
Comment (arturo-seijas) [reply httprequest_lego_provider/basicauth.py:47]: 401, line 51

**Pull request merged**